crates/vhost-device-gpio/src/backend.rs

```diff
@@ -151,9 +178,17 @@ fn start_backend<D: 'static + GpioDevice + Send + Sync>(args: GpioArgs) -> Resul
             // threads, and so the code uses unwrap() instead. The panic on a thread won't cause
             // trouble to other threads/guests or the main() function and should be safe for the
             // daemon.
-            let device = D::open(device_num).map_err(Error::CouldNotOpenDevice)?;
-            let controller =
-                GpioController::<D>::new(device).map_err(Error::CouldNotCreateGpioController)?;
+            let controller = match cfg {
+                GpioDeviceType::PhysicalDevice { id } => match D::open(id) {
+                    Ok(device) => match GpioController::<D>::new(device) {
+                        Ok(controller) => controller,
+                        Err(error) => return Err(Error::CouldNotCreateGpioController(error)),
+                    },
+                    Err(error) => return Err(Error::CouldNotOpenDevice(error)),
```

Comment:
nit: Hm... You refactor it in a later commit, but it could already be a bit shorter here with `?` use...